read anon user role from config, remove reference to public role

willgroves opened the pull request:
The get_user_roles should get the role specified for the anon user in the config.py at AUTH_ROLE_PUBLIC. Without this fix, the list of tables the anon user can see is limited to the tables visible to the "Public" role.

wgroves1 committed:
read anon user role from config, remove reference to public role

matching style

wgroves1 committed:
read anon user role from config, remove reference to public role

wgroves1 committed:
read anon user role from config, remove reference to public role

mistercrunch commented on caravel/views.py:
```diff
@@ -176,7 +176,10 @@ def check_ownership(obj, raise_if_false=True):
 def get_user_roles():
     if g.user.is_anonymous():
-        return [appbuilder.sm.find_role('Public')]
+        if config.get("AUTH_ROLE_PUBLIC",None) is not None:
```
```
public_role = config.get("AUTH_ROLE_PUBLIC")
return appbuilder.sm.find_role(public_role) if public_role else []
```

wgroves1 committed:
matching style

wgroves1 committed:
matching style

willgroves committed:
Merge branch 'master' into change_set_anon_user_group

ronbak committed:
Minor Fixes (#1484)

* Fix typo accomodate

* Fix typo calender

* Fix typo dimentions

bkyryliuk committed:
Revert "NullPool for the celery worker." (#1488)

bkyryliuk committed:
Celery uses separate db engine with NullPool. (#1492)

* Celery uses separate db engine with NullPool.

* Address comment

mistercrunch committed:
[sqllab] run only the part of the query that is selected (#1479)

mistercrunch committed:
[pep8] allowing 90 chars per line

mistercrunch committed:
[sqllab] surfacing more table metadata (indices, pk, fks) (#1485)

* Expose more table/column metadata

* [sqllab] expose more table metadata

* more tests

mistercrunch committed:
Add support for jinja templates in WHERE/HAVING clauses (#1442)

vera-liu committed:
Moved ajax call for fetching table metadata from SqlEditorLeftBar to actions (#1494)

bkyryliuk committed:
Strip sql and remove ; for csv download. (#1508)

vera-liu committed:
Moved queriesArray from render() to local state (#1505)

* Moved queriesArray from render() to local state, so that QueriesArray
is only reloaded only during switching tabs or queries object is updated.

* Changed object comparison function to take length into consideration

vera-liu committed:
Moved sqllab tests from core_tests to sqllab_tests (#1502)

* Moved sqllab tests from core_tests to sqllab_tests

* Minor changes based on comments

plumbeo committed:
Add yearly and quarterly granularities to mysql engine backend (#1518)

mistercrunch committed:
[druid] optimize Druid queries where possible (#1517)

* [druid] optimize Druid queries where possible

Trying to use timeseries, topn where possible, falling back on 2-phases
groupby only where needed

* Fixing py3 bug

Commit:
[explore V2] render all control panels and fields dynamically for each vis type (#1493)

* export functions directly rather than object at the bottom

* move viztypes to controlPanelMappings, add fieldset rows and section data

* for each viz type, render a controlPanelsContainer, controlPanelSections, FieldSetRows, and FieldsSets

* add comments, move mappings to store

* organize store and add default sections

* render all the needed sections

* add tooltip to sections

* remove console log

* use only panel panel-default, not panel-body, no need the padding

* render fields for all fields in field set

* add the rest of the control panel sections and field overrides

* fix naming

* add fieldTypes array

* don't use default section

* pass only needed state via mapStateToProps

* fix code climate errors

* linting

* move field components to their own files

* render field sets as lists

* fix field components

* use SFC

* update modal trigger test to be more accurate

* add FieldSetRow test

* add test for controlpanelsContainer

* fix test

* make code climate happy

* add freeform select field

shashanksingh committed:
Update INTHEWILD.md (#1526)

Added Faasos

mistercrunch committed:
[sqllab] templating refactor (#1504)

* Add support for jinja templates in WHERE/HAVING clauses

* Generalizing

* bugfix

bkyryliuk committed:
Support week_ending_saturday for Druid. (#1491)

* Support week_ending_saturday for Druid.

* Use period granularity

* Use ISO 8601 for period definitions.

* Fix tests

* More flexibility for the freeform choices.

Commit:
[explore-v2] control panel fixes (#1529)

* make fieldset conditions more clear

* make label required

* use render* pattern

* use slugify util for turning labels into ids

* use field rather than html

* don't need panel-title class here

Commit:
[exploreV2] remove /exploreV2 endpoint, add v2 bootstrap data to /explore endpoint (#1536)

* remove exploreV2 endpoint, add v2 bootstrap data to explore endpoint

* delete empty line

* move bootstrap data inside elif condition

Commit:
[explore-v2] handle field overrides (#1535)

* pass all props to *Field components

* s/fieldSetOverrides/fieldOverrides

* handle field overrides

vera-liu committed:
Get pivot table working in explore v2 (#1432)

* Use jquery calls in find() and html() of slice mock to
Get pivot_table viz working

* Borrowed code from caravel.js into error callback

vera-liu committed:
Added jquery methods to ChartContainer to get world_map viz working in exploreV2 (#1443)

mistercrunch committed:
[hotfix] groupby may be a set

bkyryliuk committed:
Read the user origin specification. (#1541)

mistercrunch committed:
[hotfix] druid dist_bar viz issues with non-str x values

vera-liu committed:
Added setFilter(), containerID and getFilter() to (#1360)

mock slice object in exploreV2

 - filter_box in exploreV2 was broken because containerID and
   getFilter() were not defined in the mock of slice object

mistercrunch committed:
[hotfix] remove failing Druid test

gschrader committed:
add oracle time_grains (#1544)

vera-liu committed:
Put data preview in south pane (#1486)

* Put data preview in south pane

Before: data preview of a selected table appears as a modal, but for
some cases users may want to view data and edit sql at the same time

After:
 - data preview of a selected table pops up a new tab in South Pane
 - data are saved to local state and flushed in global store in
   ResultSet component

* Moved dataPreviewId to table object

* Put back preview icon for fetching preview data

* Revert "Put back preview icon for fetching preview data"

This reverts commit b6f5dcfe64f280e26479792d99c895d38936dc1f.

* Added option to retrieve preview results after refresh

vera-liu committed:
Added scroll bar and option to collapse for Sql Editor tool bar (#1532)

* Added scroll bar and option to collapse for Sql Editor tool bar
Done:
 - Added scroll-bar to Sql Editor tool bar
 - Added hide/expand tool bar option to dropdown menu of tab

* Add more margin to give space to scroll-bar

* Add scroll to right panel independently

pinkythalli97 committed:
Update linting instructions. (#1478)

flake8 changes tests lints only the test dir.

vera-liu committed:
Added cache prop to ResultSet (#1552)

* Added cache prop to ResultSet for distiguishing query result and datapreview results

* small nit

* Only cache results.data locally

mistercrunch committed:
[druid] adding support for dimensionspecs (#1545)

More about it here:
http://druid.io/docs/latest/querying/dimensionspecs.html

fixes https://github.com/airbnb/caravel/issues/1086

vera-liu committed:
Vliu link form data explore v2 (#1540)

* Put default form_data and viz in store

* Link fields to store by firing setFormData action

* Fixed tests for Container and actions

* Moved onChange to ControlPanelsContainer, retrieve defaultFormData from fields in store

* Deleted switch statement in reducer

* Removed resetFormData and refactored setFormData in reducers

* Added text for fields

* Changed test statements

Commit:
[explore-v2] render columns based on length of fieldSets array (#1559)

* render columns based on length of fieldSets array

* fix tests

vera-liu committed:
Added action buttons to Chart Container of explore V2 (#1562)

Commit:
[explore v2] populate dynamic select field options (#1543)

* pass field options in viz json

* move field options to fetch_datasource_metadata

* on control panels container mount, fetch datasource meta data and set dynamic field choices

* render options for select fields

* use component class rather than sic

* fix linting

* fix whitespace

* delete unused var

* only render fields once datasource meta has returned

* fix typo

* add datasources and fix column formatting

* fix tests

* never used function

* fix tests

* add test for fetch_datasource_metadata

* remove unneeded props

xrmx committed:
Bump cryptography dependency to 1.5.3 (#1569)

As the 1.5 release branch builds with OpenSSL 1.1.x

mistercrunch committed:
[sqllab] bind alt+enter shortcut in AceEditor to run a query (#1554)

mistercrunch commented:
@willgroves build is broken

Commit:
add slice_name and table_name for title (#1567)

dirkkelly committed:
Link to database-urls in databaseadd view (#1480)

vera-liu committed:
Pass values from global store to fields in exploreV2 (#1561)

* Link fields to store by firing setFormData action

* Moved onChange to ControlPanelsContainer, retrieve defaultFormData from fields in store

* Pass data from store to checkbox/select/text Field

* Fixed tests

* Changed reducer back to old Object.assign() style

mistercrunch committed:
[hotfix] encode csv to utf-8

vera-liu committed:
Bind data preview tabs to sql editor (#1573)

mistercrunch committed:
Reactifying the dashboard (#1572)

vera-liu committed:
Make Sqllab a one-page app -- body not scrollable (#1551)

* Make Sqllab a one-page app -- body not scrollable

* Add scroll-container and scroll-content in main.css

mistercrunch committed:
[hotfix] resizing widgets

mistercrunch committed:
[WiP] rename project from Caravel to Superset (#1576)

* Change in files

* Renamin files and folders

* cleaning up a single piece of lint

* Removing boat picture from docs

* add superset word mark

* Update rename note in docs

* Fixing images

* Pinning datatables

* Fixing issues with mapbox-gl

* Forgot to rename one file

* Linting

* v0.13.0

* adding pyyaml to dev-reqs

mistercrunch committed:
[hotfix] datatables import issues

mistercrunch committed:
[hotfix] lint

mistercrunch committed:
Removing boat pic from README

mistercrunch committed:
v0.13.1

mistercrunch committed:
Minor documentation touchups

mistercrunch committed:
[hotfix] table view not group by without orderby fails

mistercrunch committed:
[hotfix] fix support for presto DATE and TIMESTAMP type

mistercrunch committed:
Improve Druid metadata fetching resilience (#1584)

mistercrunch committed:
[hotfix] postgres issue when slice_id is missing

geraneum committed:
Correct part_fields variable name (#1586)

mistercrunch committed:
fixing the build

vera-liu committed:
Fixed a bug with new dashboard (#1585)

kingo55 committed:
Removing broken link to old docker image (#1591)

Need to re-add if/when Kochalex updates to using Superset

vera-liu committed:
Added different Select Fields (#1583)

* Added different Select Fields
 - Switched FormGroup to react-select
 - Added multi and freeform to select, now it can take customized user
   input and insert it as options

* Fixed tests

* Small nit based on comments

mistercrunch committed:
Updating CODECLIMATE_REPO_TOKEN to new location

mistercrunch committed:
Adding Greenplum to supported dbs

mistercrunch committed:
Fix MySql time grain issue (#1590)

* Fix MySql time grain issue

* linting

* linting

mistercrunch committed:
Removing ascii_art.p from code coverage analysis

edevil committed:
Moved check to the correct place. (#1606)

mistercrunch committed:
v0.13.2

mistercrunch committed:
Changelog entries for 0.13.2

vera-liu committed:
Get query button working in explorev2 (#1581)

* Get query buttonw working in explorev2

 - Create new endpoint for updating explore viz
 - Send over new form_data when query button is pressed

* Added endpoint test

* Changes based on comments

* Added docstring for endpoint, and query spec

* Remove white space around docstring

vera-liu committed:
Add standalone to reactified dashboard page (#1596)

vera-liu committed:
Deleted unused components in exploreV2 (#1613)

* Deleted unused components in exploreV2

* Deleted constants.js

vera-liu committed:
Fixed dashboard controls for standalone bug (#1617)

mistercrunch committed:
[table viz] allow sorting on any column (#1601)

* [table viz] allow sorting on any column

* explorev2

vera-liu committed:
Vliu put datasource in store (#1610)

* Move datasource from global store object to form_data

* Moved datasource_id  and datasource_name to form_data

* Refetch defaultFormData when switching datasource

* Fixed js tests

mistercrunch committed:
More Dashboard UX unit tests (#1603)

mistercrunch committed:
[table viz] adding support for pagination (#1616)

* [table viz] adding support for pagination

* linting

Commit:
[explore-v2] add fave star and edit button to chart header (#1623)

* add fave star to chart title

* add TooltipWrapper, and css for icons

* fix linting

Commit:
add new screenshots (#1589)

mistercrunch committed:
[security] improving the security scheme (#1587)

* [security] improving the security scheme

* Addressing comments

* improving docs

* Creating security module to organize things

* Moving CLI to its own module

* perms

* Materializung perms

* progrss

* Addressing comments, linting

vera-liu committed:
Fixed bugs with viz in exploreV2 (#1609)

* Fixed bugs with viz in exploreV2

Done:
 - fix typo in pie viz
 - add metrics/groupby to dist_bar viz
 - fix typo in heatmap viz
 - add big_number_total to viz
 - fixed problem with fetching columns for datasource
 - add sqlaTimeSeries to viz
 - change row_limit and limit from number to strings so that we
 don't need to parse integers in bootstrap data

* Fix python tests

* Added order_bars checkbox for dist_bar viz

Commit:
[dashboard] give user feedback when there are unsaved changes (#1633)

* show alert and use dialog window if there are unsaved changes.

* add container class to alert

the-dcruz committed:
Make nvd3 refresh smoother. (#1618)

dodysw committed:
Update faq.rst (#1637)

Related to #1557

mistercrunch committed:
dim visualization during refresh (#1636)

Commit:
increase space between fieldsset rows (#1629)

vera-liu committed:
Fixed a bug with switching viz_type in exploreV2 (#1631)

- Issue: when switching from a viz_type outside nvd3 to a viz_type in
 nvd3, the Chart Container doesn't draw new graph
 - Fix: The reason was somehow the function inside nv.addGraph() wasn't
   called, extract the function outside and explicitly calling it solve
   the problem

vera-liu committed:
Added Alert for ControlPanel and ChartContainer (#1626)

* Added Alert for ControlPanel and ChartContainer

Done:
 - Add alert for Control Panel when fetch_datasource_metadata failes
 - Add alert for Chart Container when update_explore query fails

* Changed color to warning-yellow

* Solve linter issue

* Fixed indent and delete error_redirect

vera-liu committed:
Use Alert for visualization error (#1639)

bkyryliuk committed:
Fix alpha permission checks. (#1641)

vera-liu committed:
Save modal component for explore v2 (#1612)

* Added specs for SaveModal

* Move datasource_id and datasource_name to form_data

* Add comments

* Deleted redundant fetchDashboard

* Replcae has_key for python3

* More react and less jquery

* Added alert for save slice

* Small changes based on comments

* Use react bootstrap

Commit:
include jQuery and bootstrap (#1642)

bkyryliuk committed:
Admin / Alpha permission cleanup and fixes. (#1645)

bkyryliuk committed:
Add all_tables endpoint to allow airpal / superset perm sync. (#1614)

mistercrunch committed:
Improving jinja2 security by using SandboxedEnvironment (#1632)

http://jinja.pocoo.org/docs/dev/sandbox/#sandbox

mistercrunch committed:
User profile pages (favorites, created content, recent activity, security & access) (#1615)

* Super

* User profile page

* Fixing python style

* Python unit tests

* Touchups and js tests

* Addressing comments

kingo55 committed:
Redirect URL requests with "caravel" to "superset" (#1651)

* Redirect URL requests with "caravel" to "superset"

* Adding extra line break

bkyryliuk committed:
Programatically sync the role with user list. (#1619)

vera-liu committed:
[QuerySearch] Add loading status to QuerySearch page (#1657)

* Add loading state to QuerySearch
 * Move ajax call to ComponentDidMount
 * Show loading image during ajax call

vera-liu committed:
Fixed bug with querylink passing sql object instead of string (#1659)

vera-liu committed:
Make QueryTable scrollable in Query Search page (#1656)

vera-liu committed:
Calculate height dynamically using jquery for scrollable sqllab  (#1611)

* Calculate height dynamically using jquery for scrollable sqllab components

* Move editorHeight to App.jsx

* Calculate height dynamically for query search

vera-liu committed:
Make explore container resize with browser window (#1608)

vera-liu committed:
Temperary fix of a slice bug (#1648)

vera-liu committed:
Get sections to render when switching datasource (#1660)

* Get sections to render when switching datasource
 - Move sectionsToRender in store and use is for defaultFormData
 - Change some SelectField to FreeFormSelect according to forms.py

* Solved the css not found problem in staging

* Fixed js tests

vera-liu committed:
Rerender chart without clicking query button for fields (#1658)

* For some fields we would like to re-render chart once field is
  * changed, saving user the time to click query button
  * Such fields are stored in an array in store, could add more fields in
  * the future

mistercrunch committed:
[sqllab] adding a sql preprocessor for Presto (#1670)

* [sqllab] adding a sql preprocessor for Presto

* fixing tests

vera-liu committed:
Added filter in ControlPanelsContainer for explore V2 (#1647)

* Added filter in ControlPanelsContainer

* Move function for getting url params object to utils

* Fixed python test

* Move Filter to separate component

* Added specs and made changes based on comments

* Moved specs to right folder

xrmx committed:
docs: fix python-redis link markup (#1683)

xrmx committed:
Use smaller size for node max_old_space_size (#1679)

This makes the difference from being killed by OOM on a 8GB machine
or working.

Commit:
Missing merge_perm function. Fixes 1691. (#1692)

the-dcruz committed:
Add 'Save As' feature for dashboards (#1669)

* Add 'Save As' feature for dashboards

* Address code review comments

mistercrunch committed:
[sqllab] bugfix SouthPane doesn't update as expected (#1699)

* [sqllab] bugfix SouthPane doesn't update as expected

* Linting

mistercrunch committed:
Fixing issue #1689 (#1696)

bkyryliuk committed:
Config programmatic roles in the config.py (#1664)

* Config programmatic roles in the config.py,

* Add sql_lab.

mistercrunch committed:
[druid] fix having clause (#1694)

Commit:
make chart title larger, fix explore actions btn spacing (#1680)

Commit:
[explore-v2] fix errors on table view (#1675)

* render table name if in table view

* only render fave star and edit button if slice, not table

* fix error when table view

* use [table-name] - untitled format

* remove extra fave star

Commit:
make stack trace more readable (#1672)

* make stack trace more readable

* remove ascii_art import

* remove ascii_art.py

vera-liu committed:
Vliu explorev2 bugs (#1701)

* Fixed table_name does not exist in druid

* Make Chart container scrollable for large chart

* Fixed bug of action buttons not clickable in heatmap

* Solve codeclimate issue

* Limit overflow to x

bkyryliuk committed:
Add per schema permissions. (#1698)

* Add per schema permissions.

* Address comments.

* Add schema_access perms to the alpha and admin

* Create permissions on addition databases and datasources.

* Remove hybrid_property. Linter complains.

bkyryliuk committed:
Implement table name extraction. (#1598)

* Implement table name extraction tests.

* Address comments.

* Fix tests and reimplement the token processing.

* Exclude aliases.

* Clean up print statements and code.

* Reverse select test.

* Fix failing test.

* Test JOINs

* refactore as a class

* Check for permissions in SQL Lab.

* Implement permissions check for the datasources in sql_lab

* Address comments.

vera-liu committed:
[explorev2] Bug fixes in Save Modal (#1707)

* Bug fixes in Save Modal
Issues solved:
 - Save button doesn't pass in gotodash
 - slice_name was passed in from store as original slice_name instead of
   new one in 'saveas' action
 - datasource_type wasn't passed in to defaultViz and defaultForm
   function

* Change css filename to exploreV2

* Moved out utils

mistercrunch committed:
v0.14.0

mistercrunch committed:
Updating CHANGELOG 0.14.0

bkyryliuk committed:
Druid hotfix. (#1710)

mistercrunch committed:
v0.14.1

vera-liu committed:
[Bugfix] autocomplete in sqleditor doesnot use newly loaded table columns (#1712)

bkyryliuk committed:
Make edit / add / delete perms available to all users. (#1722)

* Make edit / add / delete perms available to all users.

* Add tests and restrict from editing the datasources.

bkyryliuk committed:
Pass schema to the select star query. (#1714)

vera-liu committed:
Redirects to login page if user not logged in at welcome page (#1723)

bkyryliuk committed:
Display full table name (schema + name) if possible. (#1728)

vera-liu committed:
[explorev2] Make chart container more responsive (#1724)

* Make chart container more responsive
 **Leave chart empty when theres error
 **Make all boolean fields auto-query chart when changed

* Replace forEach with some

* Added fields to autoQueryFields

yolken committed:
Support running superset via pex (#1713)

* Support running superset via pex

* [superset] Update default port in superset/bin/superset

* Fix codeclimate line length issues

* Fix another line length issue, in config.py

* Add trivial utils test to increase test coverage

* Clean up runserver handling

mistercrunch committed:
Adding links pointing to the new user profile page (#1704)

* Adding links pointing to the new user profile page

* Raising coverage

* Fixing tests

bkyryliuk committed:
State that npm should be between 3.9 and 4

yolken committed:
Rremove unused symlinks (#1736)

mistercrunch committed:
[sqllab] bugfix where a query has the same alias twice as output (#1734)

mistercrunch committed:
Fixing the sourcemap in dev mode (#1744)

mistercrunch committed:
[explorev2] cosmetic, smaller size for input text (#1746)

bkyryliuk committed:
Update role based on usernames not emails. (#1749)

bkyryliuk committed:
Prevent duplicated view_menu perms (#1751)

ddol committed:
Sankey Tooltip fix (#1748) (#1750)

.layer{X,Y} gives viewport offset, moved to node offset using .offset{X,Y}

mistercrunch committed:
[bugfix] bignumber comparison wrong with neg values (#1743)

* [bugfix] bignumber comparison wrong with neg values

* Handling zero div

mistercrunch committed:
[sqllab] making 'click to retrieve results' a button (#1737)

xrmx committed:
Fix superset cli for python3 (#1760)

* Fix superset cli for python3

dict.iteritems() has been removed since dict.items() returns an
iterable in python3. Shouldn't be a big deal for python2 to load
all the data into a list.

Fix #1756

* bin/superset: avoid some work when reading config

We don't need to unpack and then pack again a dictionary.

vera-liu committed:
[explorev2] fix textfield and druid bug (#1732)

* Fixed bug with textfield being empty

* Only return time grains for sqla table

vera-liu committed:
Make entire menuitem clickable for copy query (#1747)

vera-liu committed:
Enable overwrite sql in QueryHistory (#1731)

vera-liu committed:
[sqllab] Fixed js error when results are not available (#1715)

* Fixed js error when results are not available

* Flush data and query in results when running new query, keeping columns

* add exception for columns

* Move setState from componentWillMount to componentWillReceiveProps

* Nit

bkyryliuk committed:
Create users if not found. (#1753)

* Create users if not found.

* Do not fail is user is a duplicate.

* Make endpoint faster.

vera-liu committed:
Make cell-click filter in table viz optional (#1762)

* Make cell-click filter in table viz optional
 - Added table_filter checkbox in table viz
 - If set to false, clicking on a cell in table will not apply filter to
   dashboard

* Fix codeclimate issue

* Change default to false

bkyryliuk committed:
Filter table list based on the user permissions. (#1769)

* Filter table list based on the user permissions.

* Fix tests

mistercrunch committed:
[explorev2] Breaking down large files, fixing JS warnings (#1773)

* Breaking down large files, fixing JS warnings

* fix unit tests

vera-liu committed:
[WIP] Add http to copied url and move function to componentWillReceiveProps (#1780)

* Add http to copied url and move function to componentWillReceiveProps

* Added getText() to CopyToClipbaord to enable ajax calls for getting copy text

* Set ajax call to synchronous (document.execCommand only works in synchronous mode

Commit:
[ui] update logo, favicon, and new primary color (#1781)

* update favicon image

* change primary color

* update logo in header

* update logo in readme

Commit:
adjust header nav links so they are all aligned on the base line (#1786)

vera-liu committed:
Add viz thumbnails to viz_type select (#1794)

* Add viz thumbnails to viz_type select

* Replace alt with value

mistercrunch committed:
[sqllab] config item for SQLLAB_DEFAULT_DBID (#1793)

mistercrunch committed:
[hotfix] handling 0% change in big number with trendline (#1801)

vera-liu committed:
Add schema name to output column in query history (#1790)

* Add schema name to output column in query history

* Replace line break with dot between schema and table name

vera-liu committed:
Added timer to explore v2 and share it with sqllab (#1802)

* Added timer to explore v2 and share it with sqllab

* Fixed js tests

* Add timer to initial load

* Make timer smaller

* nits

mistercrunch committed:
Showing more fields in DatabaseView

darabos committed:
Add NVD3's bullet chart (#1775)

* Add NVD3's bullet chart.

* Add empty lines before nested function definitions.

* Add thumbnail for bullet chart.

* Add bullet chart to gallery.rst.

* Add "requiresTime: false", fix indentation.

* Avoid scaling bullet chart vertically.

* Use a default if no range is specified.

* Fix coloring of bullet chart.

vera-liu committed:
Add tooltips to RunAsync and CTAS button (#1792)

* Add tooltips to RunAsync and CTAS button

* Use button from components

* Phrasing

vera-liu committed:
remove extra call to get_viz in explorev2 (#1812)

* Not working errors

* Remove update_explore endpoint, only update explore endpoints in reducer on query

* Change scroll to auto and make container reponse to height:

* Remove update_explore endpoint

* Remove can_update_explore perm

mistercrunch committed:
v0.15.0

Commit:
[explore-v2] add config option for explore v2 beta users, and send through v2 path (#1671)

* add config option for explore v2 beta users, and send through v2 path

* fix long lines

* use role rather than user ids in config

* add test

* simplify role check

* remove extra  line

* use different test user for v2 tests

mistercrunch committed:
[hotfix] fixing the build

rlei committed:
Workaround for slices "Not Found" issue in IE <= 11 (#1821)

This should fix issue #1339.

IE 11 and lower has a long standing issue: out-of-document element's
pathname has no leading '/'. See

https://connect.microsoft.com/IE/feedbackdetail/view/1002846/pathname-incorrect-for-out-of-document-elements

And Superset's Slice.jsonEndpoint() method relies on pathname() to build
JSON API URL for slices:

```javascript
      jsonEndpoint() {
        const parser = document.createElement('a');
        parser.href = data.json_endpoint;
        let endpoint = parser.pathname + this.querystring();
        endpoint += '&json=true';
        endpoint += '&force=' + this.force;
        return endpoint;
      },
```

`parser` above is exactly an out-of-document element. Therefore when
running in IE <= 11, Superset would build wrong JSON endpoint URLs,
hence the 404 errors for loading data for slices.

This commit adds a simple workaround when leading '/' is missing in the
value returned by pathname().

jr-minnaar committed:
Add ADDITIONAL_MIDDLEWARE option to config (#1832)

Add documentation to explain ADDITIONAL_MIDDLEWARE

vera-liu committed:
Sort searched queries by recency (#1735)

* Sort searched queries by recency

* Fixed sqllab tests

* Add one more tr to QueryTable spec for pagination

* Change desc to asc as we reverse queries in component

vera-liu committed:
Add email-to option in action buttons for dashboard and slice (#1705)

* Add email-to option in explore action buttons

* email to option for dashboard

mistercrunch committed:
[explorev2] no bootstrap data, just metadata in exploreV2 (#1827)

json_data really just returns the *metadata* for the slice, where
get_json returns both the metadata and the data

mistercrunch committed:
[explorev2] remove unused file SqlClause.jsx (#1839)

vera-liu committed:
[sqllab] Fix sql expression bug with count distinct metrics (#1805)

* Return error message when metrics are not valid

* Fix bug with count distinct expression

* Fixed codeclimate issue

Commit:
Change default gunicorn address (#1838)

Use `0.0.0.0` as default instead of `127.0.0.1`

mistercrunch committed:
[explorev2] using a loader to load the explorev2 specific css (#1843)

bkyryliuk committed:
Permissions refactoring, optimizations and unit testing. (#1798)

* Refactor and speed up superset init

* Add unit tests.

* Test fixes.

* More test updates.

* Fix read only perms

* Address comments.

mistercrunch committed:
[explorev2] making Datasource an Viz controls not clearable (#1845)

* [explorev2] making Datasource an Viz controls not clearable

* Making choices default to empty list

mistercrunch committed:
[explorev2] making QueryAndSaveBtns disabled while running queries (#1841)

As I altered QueryAndSaveBtns to add the `disabled` prop
I also moved it to using react-boostrap

mistercrunch committed:
[explorev2] enabling redux dev tools (#1842)

vera-liu committed:
Add force_ctas_schema to query model when enabled (#1825)

* Add force_ctas_schema to query model when enabled

* Add schema to temp_table_name

* Remove extra arg in create_table_as

vera-liu committed:
Stop ChartContainer from rendering twice on chartStatus change (#1828)

* Stop ChartContainer from rendering twice on chartStatus change

* Make spinner overlay and dim chart while laoding

* Added timeout to make render more stable on resize

* Put viz in state and call resize at browser size change

* add render after height change since resize depends on render called on the same object

* Change name of renderVis to renderVisOnChange

* Only call resize at height change, persist viz in state

* Call resize wihout render at window size change

mistercrunch committed:
[explorev2] improving the scrolling/scrollbars placement (#1840)

Commit:
Use APP_ICON in template (#1855)

* use APP_ICON in template

* use APP_NAME in navbar template

vera-liu committed:
Enable freeform-select with fetched column values for filter values (#1697)

* Enable freeform-select with fetched column values for filter values
 - db migration to add filter_select_enabled
 - add freeform-multi option for Selectfield
 - modify formatFilter() function on query to accomodate filter-select

* Fix js tests

* Fix codeclimate issue

* Changes based on comments

* Add test for filter endpoint

* Extract out renderFilterFormField function from render

* Fix landscape issues

xrmx committed:
sqllab: don't hold database deletion because of query reference (#1863)

Let people delete the database even if there are sqllab queries
on that database. Instead delete them too.

Fix #1848

mistercrunch committed:
Improving database logging by adding duration, referrer and post data (#1830)

* Improving logging with duration and referrer

* Handling case where referrer is None

* Providing log_this with its own session

* Attempting to fix tests

* Fixing tests

mistercrunch committed:
[sqllab] async queries - better error handling (#1853)

wgroves1 committed:
fix build: set public role in test suite as some tests assume public user has role 'Public'

wgroves1 committed:
fix build: set public role in test suite as some tests assume public user has role 'Public'

wgroves1 committed:
Merge branch 'change_set_anon_user_group' of github.com:willgroves/caravel into change_set_anon_user_group

xrmx commented:
@willgroves please rebase your changes on top of master

wgroves1 committed:
Merge branch 'change_set_anon_user_group' of github.com:willgroves/caravel into change_set_anon_user_group

willgroves commented:
@xrmx There is a new PR #1878 with these changes rebased onto the latest master.
